components/psa/spm/COMPONENT_SPE/spm_init.c

```diff
@@ -27,6 +27,66 @@ MBED_STATIC_ASSERT( MBED_CONF_SPM_IPC_MAX_NUM_OF_MESSAGES <= PSA_HANDLE_MGR_MAX_
                     "Number of active messages exceeds maximum number of handles allowed in handles manager!"
                   );
 
+// Sanity check - memory ranges are well formed
+
+MBED_STATIC_ASSERT(
+    (PSA_SECURE_RAM_START + PSA_SECURE_RAM_SIZE - 1) > PSA_SECURE_RAM_START,
+    "Illegal secure ram region!\n"
+);
+
+MBED_STATIC_ASSERT(
+    (PSA_SECURE_FLASH_START + PSA_SECURE_FLASH_SIZE - 1) > PSA_SECURE_FLASH_START,
+    "Illegal secure flash region!\n"
+);
+
+
+MBED_STATIC_ASSERT(
+    (PSA_NON_SECURE_RAM_START + PSA_NON_SECURE_RAM_SIZE - 1) > PSA_NON_SECURE_RAM_START,
+    "Illegal non-secure ram region!\n"
+);
+
+MBED_STATIC_ASSERT(
+    (PSA_NON_SECURE_FLASH_START + PSA_NON_SECURE_FLASH_SIZE - 1) > PSA_NON_SECURE_FLASH_START,
+    "Illegal non-secure flash region!\n"
+);
+
+// Sanity check - memory ranges are not overlapping
+
+#ifndef MIN
+#define MIN( a, b ) ( ( ( a ) < ( b ) ) ? ( a ) : ( b ) )
+#endif
+
+#ifndef MAX
+#define MAX( a, b ) ( ( ( a ) > ( b ) ) ? ( a ) : ( b ) )
+#endif
+
+#define OVERLAP_CHECK( name1, start1, size1, name2, start2, size2) \
+    MBED_STATIC_ASSERT( \
+        MAX(start1, start2) > MIN((start1 + size1), (start2 + size2)), \
+        name1 " and " name2 " are overlapping!!!\n" \
+    )
+
+OVERLAP_CHECK(
+    "PSA_SECURE_RAM",     PSA_SECURE_RAM_START,     PSA_SECURE_RAM_SIZE,
+    "PSA_NON_SECURE_RAM", PSA_NON_SECURE_RAM_START, PSA_NON_SECURE_RAM_SIZE,
+);
+
+OVERLAP_CHECK(
+    "PSA_SECURE_RAM", PSA_SECURE_RAM_START, PSA_SECURE_RAM_SIZE,
+    "PSA_SHARED_RAM", PSA_SHARED_RAM_START, PSA_SHARED_RAM_START,
+);
+
+OVERLAP_CHECK(
+    "PSA_SHARED_RAM",     PSA_SHARED_RAM_START,     PSA_SHARED_RAM_START,
+    "PSA_NON_SECURE_RAM", PSA_NON_SECURE_RAM_START, PSA_NON_SECURE_RAM_SIZE,
+);
+
+OVERLAP_CHECK(
+    "PSA_SECURE_FLASH",     PSA_SECURE_FLASH_START, PSA_SECURE_FLASH_SIZE,
+    "PSA_NON_SECURE_FLASH", PSA_NON_SECURE_FLASH_START, PSA_NON_SECURE_FLASH_SIZE
+);
+
+
```

Comment:
I will go for full checks. Currently flash and ram overlap is not checked.
If we don't check it on purpose, we should add a comment with the reason for it.
Something like:
```suggestion
// Flash and RAM are assumed to be far from each other - overlap checks are not done
```